File: Arrays101/26. Remove Duplicates from Sorted Array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 1: return 1
        i = 1
        recorded = nums[0]
        #this would be a great time for a do-while loop
        for x in nums[1::]:
            if x != recorded:
                recorded = x
                nums[i] = x
                i+=1
        return i

        """
        i = 0
        recorded = {}
        for x in nums:
            if x not in recorded:
                recorded[x] = 1
                nums[i] = x
                i+=1
        return i
        """

File: Arrays101/26. Remove Duplicates from Sorted Array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length==1) return 1;
        int i = 1;

        for(int x = 1; x < nums.length; x++){
            if(nums[x] != nums[i-1]){
                nums[i] = nums[x]; //unlike python, java is faster at recalling than storing
                i++;
            }
        }
        return i;
        
    }
}